File: Easy Questions/242. Valid Anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()) {
            return false;
        }
        int s1[256] = {0};
        int t1[256] = {0};
        int size = s.length();
        for (int i = 0; i < size; i++) {
            s1[s[i]]++;
            t1[t[i]]++;
        }
        for (int i = 0; i < 256; i++) {
            if (s1[i] != t1[i]) {
                return false;
            }
        }
        return true;
    }
};

File: Easy Questions/242. Valid Anagram/valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        if len(s) != len(t):
            return False
        dics = defaultdict(int)
        for i in s:
            dics[i] += 1
        for i in t:
            if i not in dics:
                return False
            else:
                if dics[i] - 1 == 0:
                    del dics[i]
                    continue
                dics[i] -= 1
                
        return True
